proposals/2024-08-01-Self-Identification.md

```diff
@@ -0,0 +1,110 @@
+
+Placing the `self` field only in the OpenAPI Object or Arazzo Object makes it align with the existing bootstrapping process for parsing:
+
+1.  Check the `openapi` or `arazzo` field in the root OpenAPI or Arazzo Object to determine the specification version
+1.  Check the `jsonSchemaDialect` field for the default Schema Object dialect
+1.  Determine the base URI per RFC3986 §5.1.2-5.1.4 (in most cases, use the retrieval URL per §5.1.3)
+1.  ***NEW*** Check the `self` field for a base URI per RFC3986 §5.1.1; if it exists, resolve it against the base URI from the previous step and use the result as the document's actual base URI
+1.  Continue parsing the remainder of the document as usual
+
+As [OAS 3.1.1 clarifies](https://github.com/OAI/OpenAPI-Specification/pull/3758), it is already mandatory to separate location and identity for Schema Object support.
+
+Currently, associating a URI other than the current URL with a document to meet this requirement has to be done externally.  Many tools effectively by allowing the retrieval URL to be set manually, without verifying that the document actually lives at the given URL.
+
+With the new `self` field, tools need to be configured to know where to look to find the document if `self` value does not match its location.  The JSON Schema implementation linked under "Prior Art" above demonstrates several ways to accomplish this.
+
+## Detailed design
+
+This is written for the structure of the OAS, but it should be clear how it would be adapted for Arazzo.  Some amount of guidance around how to configure tools to resolve `self`-references that do not match locations probably also needs to be added in the sections on reference resolution and base URIs.
+
+```MARKDOWN
+## OpenAPI Object
+
+### Fixed Fields
+
+Field Name | Type | Description
+---|:---|:---
+self | `URI-reference` (without a fragment) | Sets the URI of this document, which also serves as its base URI in accordance with [RFC 3986 §5.1.1](https://www.rfc-editor.org/rfc/rfc3986#section-5.1.1); the value MUST NOT be the empty string and MUST NOT contain a fragment
+```
+
+## Backwards compatibility
+
+OAS 3.2 and Arazzo 1.1 documents that do not use the `self` field will behave exactly the same as OAS 3.1 and Arazzo 1.0 documents.  The change in minor version is sufficient to manage the compatibility issues, as no software that only supports up to 3.1/1.0 should attempt to parse 3.2/1.1 documents.
+
+## Alternatives considered
+
+### Plain name fragments in every Object
+
+While including `self` in every Object would produce the same complexity as JSON Schema's nested `$id`, we could just adopt an equivalent of JSON Schema's `$anchor` keyword, which (like HTML/XML's `id` attribute) creates a plain name fragment that is not tied to the location of the Object in the JSON/YAML structure.
```

Comment:
```suggestion
Including `self` in every Object would produce the same complexity as JSON Schema's nested `$id` but, that introduces unnecessary schema bloat rather than a single identifier at the root.
Adopting an equivalent of JSON Schema's `$anchor` keyword, which like (HTML/XML's `id` attribute), creates a plain name fragment that is not tied to the location of the Object in the JSON/YAML structure.
```